### PR TITLE
Bins pg interface

### PR DESCRIPTION
**Pull request by westrachel**
updated commit to add method for finding pk id of bin based on uuid; assumed we'd parse the uuid in the controller to ensure no sql injection, so I didn't add extra uuid input validation here

### CODE REVIEW

**Comment by westrachel** [comment]
actually the parameterized query prevents sql injection according to the docs: https://github.com/brianc/node-postgres/wiki/FAQ

File: models/bin.js [hunk 1]
```diff
@@ -2,3 +2,41 @@
...
+   return bin.rows[0];
+ } catch (error) {
+   console.error(error);
+ } 
+}
+
+async function newBin() {
+  try {
+    await client.query("INSERT INTO bins DEFAULT VALUES");
+    const newUUID = await client.query("SELECT uuid FROM bins ORDER BY id DESC LIMIT 1");
```
Comment (lukeoguro) [inline models/bin.js]: There might be some risk in doing this in two queries (especially if there are many people creating bins simultaneously). Looks like you can do something like the following: 

```
INSERT INTO users (name, email)
VALUES ('John Doe', 'johndoe@example.com')
RETURNING id, name;
```

What do you think?

**Review by lukeoguro — CHANGES_REQUESTED** [review]
Added a comment on the `newBin` function, but the rest looks great!

**Comment by elguaposalsero** [comment]
test

File: models/bin.js (hunk 1 above)
Comment (gcpinckert) [reply models/bin.js]: Yes this syntax works great, I used in the crud app for the individual VPS project. You can tack `RETURNING` clauses onto `UPDATE` statements too!

Comment (westrachel) [reply models/bin.js]: good callout!

**Review by lukeoguro — APPROVED** [review]
Nice!

File: models/bin.js [hunk 2]
```diff
@@ -26,9 +26,8 @@ async function binByUUID(id) {
 
 async function newBin() {
   try {
-    await client.query("INSERT INTO bins DEFAULT VALUES");
-    const newUUID = await client.query("SELECT uuid FROM bins ORDER BY id DESC LIMIT 1");
-    return newUUID.rows;
+    const newUUID = await client.query("INSERT INTO bins DEFAULT VALUES RETURNING id");
```
Comment (gcpinckert) [inline models/bin.js]: I think we actually want to return `uuid` here (the long random generated string) not `id` which is just an auto incrementing integer.

**Review by gcpinckert — CHANGES_REQUESTED** [review]
Just one small change on line 29 I think!

File: models/bin.js (hunk 2 above)
Comment (westrachel) [reply models/bin.js]: good catch, fixed!

**Review by gcpinckert — APPROVED** [review]
Looks good! Let's merge this building block and start stacking,

**Pull request merged**